falcon/api_helpers.py

```diff
@@ -121,9 +116,29 @@ def default_serialize_error(req, exception):
             preferred = 'application/xml'
 
     if preferred is not None:
+        resp.append_header('Vary', 'Accept')
```

Comment:
Awesome, this looks great! As a final step would you mind rebasing on master and squashing down to a single commit?